packages/state-transition/test/perf/util/loadState.test.ts

```diff
@@ -0,0 +1,96 @@
+
+  const seedState = stateType.deserializeToViewDU(data);
+  // cache all HashObjects
+  seedState.hashTreeRoot();
+  const config = createBeaconConfig(defaultChainConfig, seedState.genesisValidatorsRoot);
+  // TODO: EIP-6110 - need to create 2 separate caches?
+  const pubkey2index = new PubkeyIndexMap();
+  const index2pubkey: Index2PubkeyCache = [];
+  const cachedSeedState = createCachedBeaconState(seedState, {
+    config,
+    pubkey2index,
+    index2pubkey,
+  });
+
+  // TODO: precompute shufflings of state 7335360 to avoid the cost of computing shuffling
+  // as in reality we will have all shufflings
+  const shufflingGetter = (epoch: Epoch, deicisionBlock: RootHex): EpochShuffling | null => {
+    const shufflingCache = new Map<Epoch, Map<RootHex, EpochShuffling>>();
+    const currentEpoch = cachedSeedState.epochCtx.currentShuffling.epoch;
+    const previousEpoch = currentEpoch - 1;
+    const nextEpoch = currentEpoch + 1;
+    const currentEpochDecisionBlock = getShufflingDecisionBlock(seedState, currentEpoch);
+    const previousEpochDecisionBlock = getShufflingDecisionBlock(seedState, previousEpoch);
+    const nextEpochDecisionBlock = getShufflingDecisionBlock(seedState, nextEpoch);
+    shufflingCache.set(currentEpoch, new Map([[currentEpochDecisionBlock, cachedSeedState.epochCtx.currentShuffling]]));
+    shufflingCache.set(
+      previousEpoch,
+      new Map([[previousEpochDecisionBlock, cachedSeedState.epochCtx.previousShuffling]])
+    );
+    shufflingCache.set(nextEpoch, new Map([[nextEpochDecisionBlock, cachedSeedState.epochCtx.nextShuffling]]));
+    return shufflingCache.get(epoch)?.get(deicisionBlock) ?? null;
+  };
+
+  const newStateBytes = Uint8Array.from(fs.readFileSync(path.join(folder, "mainnet_state_7335360.ssz")));
+  // const stateRoot6543072 = fromHexString("0xcf0e3c93b080d1c870b9052031f77e08aecbbbba5e4e7b1898b108d76c981a31");
+  // const stateRoot7335296 = fromHexString("0xc63b580b63b78c83693ff2b8897cf0e4fcbc46b8a2eab60a090b78ced36afd93");
+  const stateRoot7335360 = fromHexString("0xaeb2f977a1502967e09394e81b8bcfdd5a077af82b99deea0dcd3698568efbeb");
+  const newStateRoot = stateRoot7335360;
+  // IMPORTANT: should not load a new separate tree (enable the code below) or the number is not correct (too bad)
+  // const newState = stateType.deserializeToViewDU(newStateBytes);
+  // startTime = Date.now();
+  // const newStateRoot = newState.hashTreeRoot();
+  // console.log("state root of state", toHexString(newStateRoot));
+  // console.log("@@@ hashTreeRoot of new state in", Date.now() - startTime, "ms");
+
+  /**
+   * My Mac M1 Pro 17:30 Sep 16 2023
+   * ✔ migrate state from slot 7335296 64 slots difference                0.4225908 ops/s    2.366355  s/op        -         14 runs   35.9 s
+   * ✔ migrate state from slot 7327776 1 day difference                   0.3415936 ops/s    2.927455  s/op        -         17 runs   52.6 s
```

Comment:
Does this mean it would be possible to support arbitrary state retrieval by archiving the full state every ~250 epochs and then deriving / loading the state from the archived state?

Reply:
this is to load unfinalized state, not finalized state, and it's not related to the archived state